pinot-core/src/main/java/org/apache/pinot/core/util/AvroUtils.java

```diff
@@ -279,6 +280,34 @@ public static void fillGenericRow(GenericRecord from, GenericRow to, Schema sche
     for (FieldSpec fieldSpec : schema.getAllFieldSpecs()) {
       String fieldName = fieldSpec.getName();
       Object avroValue = from.get(fieldName);
+      //Handle MAP types
+      if (fieldName.toUpperCase().endsWith("__KEYS")) {
+        String avroFieldName = fieldName.replaceAll("__KEYS", "");
+        Object o = from.get(avroFieldName);
+        if (o instanceof Map) {
+          Map map = (Map) o;
+          TreeSet sortedKeySet = new TreeSet<>(map.keySet());
+          Object[] keys = new Object[map.size()];
+          int i = 0;
+          for (Object key : sortedKeySet) {
+            keys[i++] = AvroUtils.transformAvroValueToObject(key, fieldSpec);
+          }
+          to.putField(fieldName, keys);
+        }
```

Comment:
Add log message or even throw an exception in `else` here? Ideally, that should never happen but if someone names their columns to have `__KEYS`, that could be caught here.

Reply:
Logging will flood the logs. Throwing exception while ingesting data is not very useful. We have metrics that will get emitted later in the code when a field is NULL.  Also, some events may not have map fields and thats perfectly valid.